### PR TITLE
[G2M] - Add Label chip component

### LINKED ISSUE #110: Components Library: Label Component
# 🎯 Usage 
- Labels are used to display content classification to users. These UI elements are often used when filtering and/or displaying search and analysis results, and generally to provide content information to users. When used with a “delete” icon, these labels can be removed, by the user.

# 🎭  Types
- **Outlined**: Labels can either have an outline, or not (by default). Outline sizes increase as label size increases. Outlines also vary in colour, based on the colour of the label used. 
Outlined
- **Text Type**: To accommodate additional use cases, labels are designed with both default text (single, bold line of text) and “with title” text (double, primary bolded, secondary regular lines of text). These differences in text are accompanied by different padding and sizing.
Text_Type
-  **Stylization for Use Case**: Labels can be stylized in different ways depending on use case. Both primary and secondary icons are optional, and can be changed depending on purpose of label. 
Stylization
-  **Shades**: Labels come in two different shades (Light & Dark), with the same five base colours (Blue, Grey, Success, Warning, Error). Light Labels are the primary use case, and should be used in most instances. Dark Labels were created for potential UI expansion, and do not currently have a use case, but may be used if stylistically necessary.

# 🫀 Anatomy
Anatomy

**Spacing, Sizing & Alignment:**
- Since labels are often stacked, or used together, it is important to leave an equal margin between tag components, other components (tag, or otherwise) and their containers. Appropriate spacing varies per use case, but in general should be no less than 8px in width and/or height from other UI elements, or other labels when stacked.
- There are three sizes for labels - small, medium, and large. These sizes have fixed heights, and wrap around label content, meaning the width can vary. Elements within labels are centre-aligned with fixed padding.
- **Small**: height: 16px;  font-size: 10pt;  icon-size: 10x10px;
Small_Anatomy
- **Medium**: height: 20px;  font-size: 12pt;  icon-size: 12x12px;
Medium_Anatomy
- **Large**: height: 24px;  font-size: 14pt;  icon-size: 14x14px; 
Large_Anatomy

# 💟 Icons Used
- [Hash](https://app.streamlinehq.com/icons/streamline-bold/interface-essential/text-formating/hash)
- [Database_Search](https://app.streamlinehq.com/icons/streamline-bold/programing-apps-websites/databases/database-search)
- [Information_Circle](https://app.streamlinehq.com/icons/streamline-bold/interface-essential/alerts/information-circle)
- [Earth](https://app.streamlinehq.com/icons/streamline-bold/maps-navigation/earth/earth-1)
- [Remove](https://app.streamlinehq.com/icons/streamline-bold/interface-essential/remove-add/remove)

# 🔗 Zeplin Link
- [Labels](https://zpl.io/nolzevx)

# 🔢  Attributes(TBD/WIP)
Property | Type | Default Value | Description
--- | --- | --- | ---
`attribute_name` | `type` | `'default_val'` | Description of the attribute

# 🔍 Design Preview
Labels

### PR DESCRIPTION
**Pull request by kc-leung**
fix #110 

- Added label component
- Added neutral-800 color

### CODE REVIEW

**Review by vlatawiec — APPROVED** [review]
Looks great!

States and interactions all match design, medium sizing still needs to be added from what I can see. Otherwise, all looks good at this stage!

**Comment by kc-leung** [comment]
> Looks great!
> 
> States and interactions all match design, medium sizing still needs to be added from what I can see. Otherwise, all looks good at this stage!

medium size is the default normal one

File: src/components/label.js [hunk 1]
```diff
@@ -6,25 +6,55 @@ import { concatStateColor, concatTargetColor } from '../utils/concat-color'
...
+  
+  return labelSize
+}
+
+const Label = forwardRef(({ classes, children, startIcon, endIcon, title, size, color, selectable, ...rest }, ref) => {
+  const labelSize = _labelSize(size)
   const borderElementsColor = concatStateColor(color, 'border', ['focus', 'hover'], [500])
   const buttonColor = concatTargetColor(color, ['bg', 'border', 'text'], [100, 100, 500])
-
+  console.log('label')
```
Comment (DoParkEQ) [inline src/components/label.js]: remove console.log

File: src/components/label.js [hunk 2]
```diff
@@ -12,49 +13,86 @@ const _labelSize = (size) => {
...
-    startIcon: `mr-5px ${classes.startIcon ? classes.startIcon : ''}`, 
-    endIcon: `ml-5px ${classes.endIcon ? classes.endIcon : ''}`,
+    startIcon: `mr-1 ${classes.startIcon ? classes.startIcon : ''}`, 
+    endIcon: `ml-1 ${classes.endIcon ? classes.endIcon : ''}`,
   })
 
   return (
     <ButtonBase ref={ref} classes={labelClasses} startIcon={startIcon} endIcon={endIcon} disabled={!selectable} {...rest}>
-      {children} <span>{title}</span>
+      {children}{title && <>: <span className={`${labelClasses.title}`}>{title}</span></>}
```
Comment (DoParkEQ) [inline src/components/label.js:188]: how do we access label class from the outside?

File: src/components/label.js [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+import { ButtonBase } from '../base-components'
+import { makeStyles } from '../utils/make-styles'
+import { getTailwindConfigColor } from '../utils/tailwind-config-color'
+
+
+const _labelSize = (size) => {
+  let labelSize = ''
+
+  switch(size) {
+  case 'lg':
```
Comment (geoerika) [inline src/components/label.js:15]: For readability, all the block code for switch should be further indented. Like here: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/switch

File: src/components/label.js [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+}
+
+const _colorType = (shade, color, focus) => {
+  let colorType = ''
+  let hoverColor = concatStateColor(color, 'bg', ['hover'], [focus ? 300 : 100])
+  let buttonColor = concatTargetColor(color, ['bg'], [focus ? 200 : 50])
+  let textColor = concatTargetColor('secondary', ['text'], [900])
+
+  switch(shade) {
+  case 'light':
```
Comment (geoerika) [inline src/components/label.js:50]: See obs above.

File: src/components/label.js [hunk 5]
```diff
@@ -0,0 +1,222 @@
...
+  
+  return colorType
+}
+
+const _iconHoverColor = (shade, color) => {
+  let hoverColor = ''
+  let shadeVal = '600'
+
+  switch(shade) {
+  case 'light':
```
Comment (geoerika) [inline src/components/label.js:116]: See above!

File: src/components/label.js [hunk 6]
```diff
@@ -0,0 +1,222 @@
...
+  return hoverColor
+}
+
+const customClasses = (shade, color) => {
+  const iconHoverColor = _iconHoverColor(shade, color)
+
+  return makeStyles({
+    buttonContainer: {
+      borderWidth: 0,
+
```
Comment (geoerika) [inline src/components/label.js:147]: Unnecessary empty line.

File: src/components/label.js [hunk 7]
```diff
@@ -0,0 +1,222 @@
...
+      '& .icon-hover': {
+        '&:hover': {
+          backgroundColor: iconHoverColor,
+        },
+      },
+    },
+  })
+}
+
+const Label = forwardRef(({ classes, children, startIcon, endIcon, title, size, shade, color, onClickStartIcon, onClickEndIcon, disabled, ...rest }, ref) => {
```
Comment (geoerika) [inline src/components/label.js:157]: This is a very long line, I would list all the props on different lines.

**Review by geoerika — APPROVED** [review]
Minor comments.

File: src/components/label.js (hunk 2 above)
Comment (kc-leung) [reply src/components/label.js:188]: you can access each of the main container label classes passing down a prop in this way:
```
classes: { 
    root: 'custom-root', 
    content: 'custom-container', 
    title: 'custom-title', 
    startIcon: 'custom-start-icon', 
    endIcon: 'custom-end-icon' 
  }
```

File: src/components/label.js (hunk 3 above)
Comment (kc-leung) [reply src/components/label.js:15]: it is from lint --fix tho do you know how I can make not complaining about it?

**Pull request merged**